Add tracking for amulet of avarice to skull timer

juusokarjanlahti opened the pull request:
I re-skull using Amulet of Avarice at the bank and lose track when the skull will expire.

The plugin only tracks skull timers acquired from Emblem Traders. I wanted to also track the timer for Amulet of Avarice.

When player equips Amulet of Avarice the skull timer infobox is set to 20 minutes, when it's unequipped it starts counting down.

![image](https://github.com/user-attachments/assets/2eb3f375-1509-4f77-bdb2-089b2adac335)

https://github.com/user-attachments/assets/5655f68c-25e0-4878-97f0-9c83d6494939

juusokarjanlahti committed:
Add amulet of avarice timer

Teekiz commented:
Hey, thanks for submitting this. I've taken a little bit of a look but it seems that there may be an issue causing the timer to be re-created once expiring after obtaining a skull through the amulet. I think it may have to do with how your if block between lines 144 - 148. I'd be happy to take a bit of a closer look when I get a minute.
![skulltimerpullreq](https://github.com/user-attachments/assets/eb647766-fc53-45f1-9eb2-2d975bb6c58d)

Teekiz committed:
Refactored SkulledTimerPlugin to optimize equipment checks:

- Replaced on every game tick check with onItemContainerChanged event to avoid log and timer spam.
- Moved `checkAmuletOfAvarice` logic to a new `EquipmentChecker` class.
- Split existing logic into `hasEquipmentChanged` and `isWearingSkulledItem` for clarity and maintainability.
- Introduced `SkulledItems` enum for better future extensibility.
- Added javadoc comments for better code clarity.

Co-authored-by: Juuso Karjanlahti <98909830+juusokarjanlahti@users.noreply.github.com>

Teekiz commented:
Hey sorry for taking so long to make these changes but I've modified the code and split it into its own separate equipment class (I plan to add further features) so I figured it would make it a bit more maintainable.

I've split your existing checkAmuletOfAvarice method into the new class and separated between the two methods (and added support for the cape of skulls). I've also added Runelite's onItemContainerChanged event, this should fix the problem where the timer was being re-added, but it should also stop the timer from being recreated every time it's changed. 

If you get the chance, could you give it a test and see if it matches with your suggestion (and if there are any issues with it)? Thanks
